### PR TITLE
 [SW-106]  Accounts card

### PR DESCRIPTION
**Pull request by subhod-i**

### CODE REVIEW

**Comment by subhod-i** [comment]
Adding tests.

File: src/popup/router/components/AccountCard.vue [hunk 1]
```diff
@@ -1,42 +1,67 @@
...
+          {{ $t('pages.fungible-tokens.tokens') }}
+        </span>
+      </div>
+      <div class="buttons">
+        <RouterLink :to="{ name: 'name-claim' }">
+          <ReceiveIcon :style="iconCssProps" />
+        </RouterLink>
+        <RouterLink :to="{ name: 'name-claim' }">
+          <SendIcon :style="iconCssProps" />
+        </RouterLink>
```
Comment (pstachel) [inline src/popup/router/components/AccountCard.vue]: Should those two RouterLinks redirect to the same route?

File: src/popup/router/components/AccountInfo.vue [hunk 2]
```diff
@@ -1,154 +1,65 @@
...
-        </ButtonPlain>
-        <RouterLink
-          v-if="UNFINISHED_FEATURES && $route.path !== '/accounts'"
-          to="/accounts"
-        >
-          <Settings />
-        </RouterLink>
-      </div>
-    </div>
+  <div :class="['account-info']">
```
Comment (pstachel) [inline src/popup/router/components/AccountInfo.vue:2]: it could be simplified to 

```
<div class="account-info">
```

File: src/popup/router/components/AccountInfo.vue [hunk 3]
```diff
@@ -1,154 +1,65 @@
...
+      <div class="account-details">
+        <template v-if="!copied">
+          <Truncate
+            v-if="accounts[idx].name"
+            :str="accounts[idx].name"
+          />
+          <span
+            v-else
+            data-cy="account-number"
+            class="account-name"
```
Comment (pstachel) [inline src/popup/router/components/AccountInfo.vue:18]: should those two be different?

File: src/popup/router/components/AccountInfo.vue [hunk 4]
```diff
@@ -185,142 +84,67 @@ export default {
...
+        display: flex;
+        opacity: 0.85;
+        margin-top: 8px;
+        letter-spacing: -1px;
 
-      &:not(.text) {
-        border-bottom: 1px dashed variables.$color-blue;
-        margin: 0 8px;
+        &:hover {
+          color: variables.$color-white;
```
Comment (pstachel) [inline src/popup/router/components/AccountInfo.vue:127]: it doesn't have to be repeated - color is declared in the class

File: src/popup/router/components/AccountInfo.vue [hunk 5]
```diff
@@ -185,142 +84,67 @@ export default {
...
-  }
-
-  .copied {
-    display: flex;
-    align-items: center;
-
-    span {
-      width: 100%;
+      .ae-address {
+        @extend %face-mono-14-medium;
```
Comment (pstachel) [inline src/popup/router/components/AccountInfo.vue:119]: empty line is needed after `@extend ...`

File: src/popup/router/components/AccountInfo.vue [hunk 6]
```diff
@@ -185,142 +84,67 @@ export default {
...
 
-      &.text {
-        white-space: nowrap;
-        color: variables.$color-blue;
+      .copied {
+        align-items: center;
+        justify-content: center;
+        display: flex;
+        flex: 0.9;
+        border: dashed 1.1px #FFF;
```
Comment (pstachel) [inline src/popup/router/components/AccountInfo.vue:137]: It won't work - 1.1px will be displayed same as 1px

Comment (pstachel) [inline src/popup/router/components/AccountInfo.vue:137]: `#FFF` should be changed to `variables.$color-white`

File: src/popup/router/components/AccountInfo.vue [hunk 7]
```diff
@@ -185,142 +84,67 @@ export default {
...
+        border-radius: 5px;
 
-        @extend %face-sans-14-regular;
+        svg {
+          margin-right: 6px;
+        }
 
-        line-height: 16px;
+        .text {
+          @extend %face-sans-16-medium;
```
Comment (pstachel) [inline src/popup/router/components/AccountInfo.vue:145]: empty line needed after `@extend`

File: src/store/getters.js [hunk 8]
```diff
@@ -69,7 +65,7 @@ export default {
     return networks[network];
   },
   getProfileImage: (_, { activeNetwork }) => (address) => `${activeNetwork.backendUrl}/profile/image/${address}`,
-  getAvatar: () => (address) => `${AVATAR_URL}${address}`,
+  getAvatar: () => (address) => `${AVATAR_URL}${address}}`,
```
Comment (pstachel) [inline src/store/getters.js:68]: I think there should be no curly brace at the end

**Review by pstachel — CHANGES_REQUESTED** [review]

**Comment by CedrikNikita** [comment]
the fixup commit contatins of this changes:
- adjust border radius of a card
- use proper tags
- removed duplicate typography variables
- adjusted weight and font-sizes
- add letter-spacing for truncatedAddress
- addjusted AeBalance component height and margins
- remove link from a Account ${number} to a Register name page, not a link anymore
- removed unused logic with customNames, shift of a background, colors
- removed unused svgs
- set actual links to the Transfer and Receive pages
- set a cursor pointer for the whole card

File: src/popup/router/components/AeBalance.vue [hunk 9]
```diff
@@ -26,16 +25,25 @@ export default {
...
+  }
+
+  .token-integer {
+    @extend %face-sans-30-bold;
+  }
+
+  .token-fractional {
+    @extend %face-sans-24-bold;
+
+    line-height: 32px;
```
Comment (subhod-i) [inline src/popup/router/components/AeBalance.vue:45]: How do you pick the line heights?

File: src/store/modules/accounts/hdWallet.js [hunk 10]
```diff
@@ -15,16 +15,7 @@ export default {
...
-        {
-          idx: state.nextAccountIdx,
-          color:
-            // eslint-disable-next-line no-bitwise
-            state.nextAccountIdx === 1 ? '#00FF9D' : `#${((Math.random() * 0xffffff) << 0).toString(16)}`,
-          shift: Math.floor(Math.random() * 100),
-          type,
-        },
-        { root: true });
+      commit('accounts/add', { idx: state.nextAccountIdx, type }, { root: true });
```
Comment (subhod-i) [inline src/store/modules/accounts/hdWallet.js:18]: why these changes are relevant?

File: src/store/getters.js [hunk 11]
```diff
@@ -69,7 +65,7 @@ export default {
     return networks[network];
   },
   getProfileImage: (_, { activeNetwork }) => (address) => `${activeNetwork.backendUrl}/profile/image/${address}`,
-  getAvatar: () => (address, color) => `${AVATAR_URL}${address}?color=${encodeURIComponent(color)}`,
+  getAvatar: () => (address) => `${AVATAR_URL}${address}`,
```
Comment (subhod-i) [inline src/store/getters.js:68]: Sorry! I was supposed to remove this.

File: src/popup/router/components/BalanceInfo.vue [hunk 12]
```diff
@@ -3,43 +3,20 @@
     class="balance-info"
     data-cy="balance-info"
   >
-    <div class="balance-wrapper">
-      <AeBalance :balance="Number(balances[idx])" />
```
Comment (subhod-i) [inline src/popup/router/components/BalanceInfo.vue]: we did not incorporate currency change settings.

Comment (CedrikNikita) [reply src/popup/router/components/BalanceInfo.vue]: They should not be part of a card anymore.

File: src/store/modules/accounts/hdWallet.js (hunk 10 above)
Comment (CedrikNikita) [reply src/store/modules/accounts/hdWallet.js:18]: We've removed the previous logic with `shift` and `color` being generated for each account and stored, now we can not store it.

Comment (subhod-i) [reply src/store/modules/accounts/hdWallet.js:18]: tobe specific, why `commit` method is used in here?

**Review by CedrikNikita — CHANGES_REQUESTED** [review]
The commit history must be clean in order to merge.

File: src/store/modules/accounts/hdWallet.js (hunk 10 above)
Comment (CedrikNikita) [reply src/store/modules/accounts/hdWallet.js:18]: `The only way to actually change state in a Vuex store is by committing a mutation.`
https://vuex.vuejs.org/guide/mutations.html

File: src/popup/router/components/AeBalance.vue (hunk 9 above)
Comment (CedrikNikita) [reply src/popup/router/components/AeBalance.vue:45]: According to figma file.

**Comment by Liubov-crypto** [comment]
if I choose a long name, then the darkening of the borders of the name will be visible on the account card. Is this ok, @paolomolo? 

зат

**Comment by paolomolo** [comment]
> if I choose a long name, then the darkening of the borders of the name will be visible on the account card. Is this ok, @paolomolo?

Truncation of long .chain-names should be improved in another story. So it is ok like that for now. Thanks

**Review by pstachel — APPROVED** [review]

**Comment by CedrikNikita** [comment]
This will also speed up the initialization.
